experiment/run_experiment.py

```diff
@@ -66,14 +66,17 @@ def read_and_validate_experiment_config(config_filename: str) -> Dict:
     and returns it."""
     config = yaml_utils.read(config_filename)
     filestore_params = {'experiment_filestore', 'report_filestore'}
-    cloud_config = {'cloud_compute_zone'}
-    string_params = cloud_config.union(filestore_params)
+    cloud_config = {'cloud_compute_zone', 'cloud_project'}
+    local_config = {'docker_registry'}
```

Comment:
`docker_registry` will be used for local and cloud too, no? why `local_config =`?

Reply:
Here means, `docker_registry` is required for local.
For cloud, since `cloud_project` is required and then `docker_registry` is not required.

Reply:
Probably add a comment on this below, that "if not local_experiment"

Reply:
Done. Add a TODO comment above it.

Reply:
Basically, we can accept docker_registry for both settings, but only required in local setting.

Maybe in the future, we can make it required in both settings, especially when we know whether `cloud_project` can be removed or not. If `cloud_project` can be removed for cloud running, then it will be a good time to make `docker_registry` required for cloud setting, too.

Reply:
I think the question whether `cloud_project` can be removed or not is orthogonal to the question of making `docker_registry` a mandatory common option for all platforms. The benefit of making `docker_registry` a common option - instead of deriving it in the cloud case - would be simplicity: 

(1) It'd be clear in all cases what it is set to (ie. to the value in the config file), and 

(2) we wouldn't need these additional branches in the code here and below (like `if local_experiment:` and `if not local_experiment and 'docker_registry' not in config:`).

At this point of the project, in my opinion, simplicity is higher priority than backwards compatibility.

wdyt?

Reply:
I'm fine for this suggestion. Let me change it accordingly.